### PR TITLE
[Phase 1 Remove usage of Stream/Optional on Method Return Value from the interface AdaptiveMedia]Lps 172879 adaptive media 2 17

### PR DESCRIPTION
**Pull request by slnn**
https://issues.liferay.com/browse/LPS-172879

This pull aims to remove the Option usage in this method of interface AMImageAttributeMapping

getValueOptional(AMAttribute<AMImageProcessor, V> amAttribute)

### CODE REVIEW

**Comment by slnn** [comment]
ci:test:lima

**Comment by Kvale1733** [comment]
@slnn No regressions found.

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/AMImageRequestHandler.java [hunk 1]
```diff
@@ -276,22 +277,21 @@ private void _processAMImage(
...
-		}
+		String adaptiveMediaConfigurationUuid = adaptiveMedia.getValue(
+			AMAttribute.getConfigurationUuidAMAttribute());
 
 		String attributeMappingConfigurationUuid =
 			amImageAttributeMapping.getValue(
 				AMAttribute.getConfigurationUuidAMAttribute());
 
-		if (adaptiveMediaConfigurationUuid.equals(
+		if ((adaptiveMediaConfigurationUuid == null) &&
```
Comment (dantewang) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/handler/AMImageRequestHandler.java:287]: @slnn Could you try to see if it's possible to simplify this?

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/util/comparator/AMAttributeDistanceComparator.java [hunk 2]
```diff
@@ -65,14 +65,18 @@ public long compare(
...
-			Optional<?> valueOptional2 = adaptiveMedia2.getValueOptional(
-				amAttribute);
-
-			Optional<Long> valueOptional3 = valueOptional1.flatMap(
-				value1 -> valueOptional2.map(
-					value2 -> amAttribute.compare(value1, value2)));
+			Object value1 = adaptiveMedia1.getValue(amAttribute);
+			Object value2 = adaptiveMedia2.getValue(amAttribute);
+
+			Optional<Long> valueOptional3;
```
Comment (dantewang) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/util/comparator/AMAttributeDistanceComparator.java:71]: @slnn Let's clean this up as long as it's method internal and related to the logic you modified.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by slnn** [comment]
Resend in https://github.com/liferay-core-infra/liferay-portal/pull/3725

**Pull request closed without merging**